### DISCUSSION TITLE
feature / bug: update the component after component.set('type')

### DISCUSSION THREAD

**Discussion by aliibrahim123** [GrapesJS/grapesjs#3805] [Ideas]
when i was working on a trait that change the type of the component,
i noticed that you must update the component using this script :
```
dgh = editor.getSelected().toHTML();
	dgh1 = dgh.split(' ', 1)[0];
	dgh2 = dgh.replace(`${dgh1} `, '');
	dgh = `${dgh1} data-gjs-type = '${editor.getSelected().get('type')}' ${dgh2}`;
	editor.select(editor.getSelected().replaceWith(dgh));
```
and this is also for **component.set('type')**
so @artf can you make that if we change the type of a component the component update automatically

**Comment by artf** [comment]
It doesn't make sense, from the core point of view, to have such a listener. Each type can have its own properties and behaviors, so if you need such functionality, add it to your component, there is no need to have it in the core.
